crates/proof-impl/btc-blockspace/src/block.rs

```diff
@@ -128,53 +156,87 @@ pub fn check_pow(block: &Header) -> bool {
 
 #[cfg(test)]
 mod tests {
-    use bitcoin::{hashes::Hash, TxMerkleNode, WitnessMerkleNode};
-    use rand::{rngs::OsRng, Rng};
-    use strata_state::{l1::generate_l1_tx, tx::ProtocolOperation};
-    use strata_test_utils::{bitcoin::get_btc_mainnet_block, ArbitraryGenerator};
+    use bitcoin::Witness;
+    use strata_test_utils::bitcoin::{get_btc_chain, get_btc_mainnet_block};
+
+    use super::*;
 
-    use super::compute_merkle_root;
-    use crate::block::{check_pow, check_witness_commitment, compute_witness_root};
+    #[test]
+    fn test_block_with_valid_witness() {
+        let block = get_btc_mainnet_block();
+        let coinbase_inclusion_proof = L1TxProof::generate(&block.txdata, 0);
+        assert!(check_integrity(&block, &coinbase_inclusion_proof));
+    }
 
     #[test]
-    fn test_tx_root() {
+    #[should_panic]
+    fn test_block_with_invalid_coinbase_inclusion_proof() {
         let block = get_btc_mainnet_block();
-        assert_eq!(
-            block.compute_merkle_root().unwrap(),
-            TxMerkleNode::from_byte_array(*compute_merkle_root(&block).unwrap().as_ref())
-        );
+        let empty_inclusion_proof = L1TxProof::new(0, vec![]);
+        assert!(check_integrity(&block, &empty_inclusion_proof));
     }
 
     #[test]
-    fn test_wtx_root() {
+    #[should_panic]
```

Comment:
Since `check_integrity` returns a bool, it would be better to explicitly check it returns false rather than expecting a panic to occur which can occur for totally different reason, even though that might not be the case here.